UnitTests/ArrayHelperTests.cs

```diff
@@ -0,0 +1,321 @@
+            new object[] { new[] { 101, 5, 9, 73, 16, 100, 10, -178}, 0 },
+            new object[] { new[] { -5, -9, -73, 0, -19, -1, -10, 3}, -7 },
+        };
+
+        [TestCaseSource(nameof(FindSumOddIndexElementsSource))]
+        public static void FindSumOddIndexElements_WhenArrayNotEmpty_ShouldFindMinElement(int[] array, int expectedResult)
+        {
+            int actualResult = FindSumOddIndexElements(array);
+
+            Assert.AreEqual(expectedResult, actualResult);
+        }
+
+        [Test]
+        public static void FindSumOddIndexElements_WhenArrayIsEmpty_ShouldThrowException()
+        {
+            try
+            {
+                FindSumOddIndexElements(new int[] { });
+            }
+            catch (ArgumentException ex)
+            {
+                Assert.AreEqual("Array is empty", ex.Message);
+                Assert.Pass();
+            }
+
+            Assert.Fail("No throw exception");
+        }
+
+        static object[] ReverseSource = new[]
+        {
+            new object[] { new[] { -9, 5, -4, 73, 19, 0, -1, 100 }, new[] { 100, -1, 0, 19, 73, -4, 5, -9 } },
+            new object[] { new[] { 101, 5, 9, 73, 16, 100, 10, -178 }, new[] { -178, 10, 100, 16, 73, 9, 5, 101 } },
+            new object[] { new[] { -5, -9, -73, 0, -19, -1, -10, 3, -7 }, new[] {-7, 3, -10, -1, -19, 0, -73, -9, -5 }}
+        };
+
+        [TestCaseSource(nameof(ReverseSource))]
+        public static void Reverse_WhenArrayNotEmpty_ShouldFindMinElement(int[] array, int[] expectedResult)
+        {
+            int[] actualResult = Reverse(array);
+
+            Assert.AreEqual(expectedResult, actualResult);
+        }
+
+        [Test]
+        public static void Reverse_WhenArrayIsEmpty_ShouldThrowException()
+        {
+            try
+            {
+                Reverse(new int[] { });
+            }
+            catch (ArgumentException ex)
+            {
+                Assert.AreEqual("Array is empty", ex.Message);
+                Assert.Pass();
+            }
+
+            Assert.Fail("No throw exception");
+        }
+
+        static object[] CalculateSumOddElementsSource = new[]
+        {
+            new object[] { new[] {-9, 5, -4, 73, 19, 0, -98, 100}, 88 },
+            new object[] { new[] { 101, 5, 9, 73, 16, 100, 10, -178}, 188 },
+            new object[] { new[] { -4, -9, -74, 0, -19, -1, -10, 3}, -26 },
+        };
+
+        [TestCaseSource(nameof(CalculateSumOddElementsSource))]
+        public static void CalculateSumOddElements_WhenArrayNotEmpty_ShouldFindMinElement(int[] array, int expectedResult)
+        {
+            int actualResult = CalculateSumOddElements(array);
+
+            Assert.AreEqual(expectedResult, actualResult);
+        }
+
+        [Test]
+        public static void CalculateSumOddElements_WhenArrayIsEmpty_ShouldThrowException()
+        {
+            try
+            {
+                CalculateSumOddElements(new int[] { });
```

Comment:
I don't think that empty array for sum should throw ex. I think null array should throw ex but not empty

Reply:
fixed